### ISSUE TITLE
C# evaluate with GPU on GPU trained model

### ISSUE THREAD

**Issue by harewei** [Microsoft/CNTK#1100]
Basically I ran into the same thing as #999 , where I'm unable to evaluate a GPU trained modified ResNet model.  I tried to follow [ https://github.com/Microsoft/CNTK/blob/master/Tests/EndToEndTests/EvalClientTests/CSEvalClientTest/CSEvalClientTest.csproj](url)
where I copy and paste 
```
<ItemGroup>
    <ProjectReference Include="..\..\..\..\Source\Extensibility\EvalWrapper\EvalWrapper.vcxproj">
      <Project>{ef766cae-9cb1-494c-9153-0030631a6340}</Project>
      <Name>EvalWrapper</Name>
    </ProjectReference>
</ItemGroup>
```
into my C# project .csproj file, but it was not working (even after I fix the project reference directory).

Is there another way to fix this problem?  Or perhaps the NuGet package will be updated at some point to support GPU?

As a small side question, if I train the network with GPU, would I be able to evaluate on a machine that has no GPU installed?

**Comment by zhouwangzw** [comment]
@harewei What is the error message for "not woking"? 

Yes, in general you are able to evaluate a model trained with GPU on a CPU-Only machine. For ResNet model, the model should be trained using CNTK BatchNormalization Engine. (setting useCntkEngine to true in the line BatchNormalizationLayer {spatialRank = 2, ....} of Macro.bs). Then you can evaluate the ResNet also on a CPU device. 
We are currently working on the new version of C# Eval which will support eval on GPU device.
Thanks.

**Comment by harewei** [comment]
I actually used ResNet which gave me this CPU only error, but when using ConvNet it doesn't (even though I also trained with GPU with deviceID=0).  I will post the error later since I need to recreate it first.

**Comment by zhouwangzw** [comment]
@harewei This only happens when your model has BatchNormalizationLayer using CuDNN BN engine. Your ConvNet probably does not use Batch Normalization? 
Thanks.

**Comment by zhouwangzw** [comment]
The release CNTK2.0-Beta8 or onwards now supports to evaluate the model with BN on CPU, which is trained on GPU. The BN engine has now a fallback path from cuDNN engine to CNTK Engine if GPU is not available.

**Comment by zhouwangzw** [comment]
I am closing this issue. Please feel free to reopen if it needed.
